### PR TITLE
aws-vault: add wrapper and simple install check

### LINKED ISSUE #121154: aws-vault: missing xdg-open
**Describe the bug**
Due to missing dependency, browser page never opens during `login`

**To Reproduce**
1. `❯ aws-vault login test_profile`

**Expected behavior**
Browser page must open and authentication process finish.

**Screenshots**

**Additional context**
```bash
❯ aws-vault login test_profile
Opening the SSO authorization page in your default browser (use Ctrl-C to abort)
https://<sso_URL>.amazonaws.com/?user_code=<CODE>
```

**Notify maintainers**

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
```bash
 - system: `"x86_64-linux"`
 - host os: `Linux 5.11.14, NixOS, 21.05pre285574.8e4fe32876c (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210326_dd77f71`
 - channels(ymatsiuk): `""`
 - channels(root): `"nixos-21.05pre285574.8e4fe32876c"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: aws-vault
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by ymatsiuk**
###### Motivation for this change
Closes #121154 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/admin/aws-vault/default.nix [hunk 1]
```diff
@@ -12,9 +18,10 @@ buildGoModule rec {
 
   vendorSha256 = "sha256-Lb5iiuT/Fd3RMt98AafIi9I0FHJaSpJ8pH7r4yZiiiw=";
 
-  nativeBuildInputs = [ installShellFiles ];
+  nativeBuildInputs = [ installShellFiles makeWrapper ];
 
   postInstall = ''
+    wrapProgram $out/bin/aws-vault --prefix PATH : ${xdg-utils}/bin
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/aws-vault/default.nix:24]:
```suggestion
    wrapProgram $out/bin/aws-vault --prefix PATH : ${lib.makeBinPath [ xdg-utils ]}
```

File: pkgs/tools/admin/aws-vault/default.nix [hunk 2]
```diff
@@ -32,6 +39,12 @@ buildGoModule rec {
     -X main.Version=v${version}
   '';
 
+  doInstallCheck = true;
+
+  installCheckPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/aws-vault/default.nix:44]: I read something about this should now be in passthru tests but I am not sure.

Comment (ymatsiuk) [reply pkgs/tools/admin/aws-vault/default.nix:44]: This is pretty far from anything we call test 🙂 Just a sanity check.

Where can I find more info on the agreements/conventions?

Comment (zimbatm) [reply pkgs/tools/admin/aws-vault/default.nix:44]: @SuperSandro2000 this only applies to tests that take a long time to run. The idea is that for integration tests a second derivation is created that runs it, and doesn't block for the main derivation to build and produce an output. Here is should only add a few milliseconds so it's not worth the overhead of another derivation.

@ymatsiuk search for `passthru.tests` in the code base. It looks like most of them are using NixOS tests.

File: pkgs/tools/admin/aws-vault/default.nix [hunk 3]
```diff
@@ -32,6 +39,12 @@ buildGoModule rec {
...
+    $out/bin/aws-vault --version 2>&1 | grep ${version} > /dev/null
```
Comment (zimbatm) [inline pkgs/tools/admin/aws-vault/default.nix:45]: Does this test that `xdg-utils` is being found?

An even better version would be to add `PATH=/not/exist ` before the command to make sure it doesn't source anything from the PATH.

Comment (ymatsiuk) [reply pkgs/tools/admin/aws-vault/default.nix:45]: Hmm, it doesn't actually test `xdg-utils`, just a simple sanity check to make sure that version is printed and binary is executable 🙂 . It's not necessary though.
The `xdg-utils` dependency is [implicit](https://github.com/99designs/aws-vault/blob/0c10c45693116608fb3765be304f554189f0df01/cli/login.go#L137). To properly test it we would need to have credentials 🤷🏻

File: pkgs/tools/admin/aws-vault/default.nix (hunk 2 above)
Comment (ymatsiuk) [reply pkgs/tools/admin/aws-vault/default.nix:44]: Even though I see a use case for proper tests, this PR is not covering those. This is just a simple install check. I can also remove if you think this is not the best place for it.

**Pull request merged**